sdk/src/Services/DynamoDBv2/Custom/DataModel/_bcl/Context.Sync.cs

```diff
@@ -26,211 +26,148 @@ public partial class DynamoDBContext : IDynamoDBContext
         public void Delete<T>(object hashKey, DynamoDBOperationConfig operationConfig)
         {
-            DeleteHelper<T>(hashKey, null, operationConfig);
+            DeleteHelper<T>(hashKey, null, new DynamoDBFlatConfig(operationConfig, Config));
         }
 
-        /// <summary>
-        /// Deletes an item in DynamoDB corresponding to a given hash-and-range primary key.
-        /// 
-        /// No version check is done prior to delete.
-        /// Type must be marked up with DynamoDBTableAttribute and at least
-        /// one public field/property with DynamoDBHashKeyAttribute.
-        /// </summary>
-        /// <typeparam name="T">Type of object.</typeparam>
-        /// <param name="hashKey">Hash key element of the object to delete.</param>
-        /// <param name="rangeKey">Range key element of the object to delete.</param>
+        /// <inheritdoc/>
+        [Obsolete("Use the Delete overload that takes LoadConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to Delete.")]
+        public void Delete<T>(object hashKey, DeleteConfig deleteConfig)
+        {
+            DeleteHelper<T>(hashKey, null, new DynamoDBFlatConfig(deleteConfig?.ToDynamoDBOperationConfig(), Config));
+        }
+
+        /// <inheritdoc/>
         public void Delete<T>(object hashKey, object rangeKey)
         {
-            DeleteHelper<T>(hashKey, rangeKey, null);
+            DeleteHelper<T>(hashKey, rangeKey, new DynamoDBFlatConfig(null, Config));
         }
 
-        /// <summary>
-        /// Deletes an item in DynamoDB corresponding to a given hash-and-range primary key.
-        /// 
-        /// No version check is done prior to delete.
-        /// Type must be marked up with DynamoDBTableAttribute and at least
-        /// one public field/property with DynamoDBHashKeyAttribute.
-        /// </summary>
-        /// <typeparam name="T">Type of object.</typeparam>
-        /// <param name="hashKey">Hash key element of the object to delete.</param>
-        /// <param name="rangeKey">Range key element of the object to delete.</param>
-        /// <param name="operationConfig">Config object which can be used to override that table used.</param>
+        /// <inheritdoc/>
+        [Obsolete("Use the Delete overload that takes LoadConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to Delete.")]
         public void Delete<T>(object hashKey, object rangeKey, DynamoDBOperationConfig operationConfig)
         {
-            DeleteHelper<T>(hashKey, rangeKey, operationConfig);
+            DeleteHelper<T>(hashKey, rangeKey, new DynamoDBFlatConfig(operationConfig, Config));
+        }
+
+        /// <inheritdoc/>
+        [Obsolete("Use the Delete overload that takes LoadConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to Delete.")]
```

Comment:
This shouldn't be marked as obsolete since this is the new API

Reply:
Oops, too trigger happy with copy+paste. Removed in https://github.com/aws/aws-sdk-net/pull/3387/commits/1e8128085c97ee7a5f6d0803be53f48078bd4e05